client/consul.go

```diff
@@ -100,15 +124,73 @@ func (c *ConsulClient) GetLicense(ctx context.Context, q *consulapi.QueryOptions
 	var license string
 
 	f := func(context.Context) error {
+		var err error
 		license, err = c.Operator().LicenseGetSigned(q)
+
+		// Process the error by wrapping it in the correct error types
```

Comment:
I thought about separating this processing logic it's own function, but I think the interpretation of `404` in this case might not be universal. For now just leaving the logic in place, we can re-evaluate as we add more supported API calls